src/main/java/org/graylog/integrations/notifications/types/SlackEventNotificationConfig.java

```diff
@@ -0,0 +1,180 @@
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import org.graylog.events.contentpack.entities.EventNotificationConfigEntity;
+import org.graylog.events.event.EventDto;
+import org.graylog.events.notifications.EventNotificationConfig;
+import org.graylog.events.notifications.EventNotificationExecutionJob;
+import org.graylog.scheduler.JobTriggerData;
+import org.graylog2.contentpacks.EntityDescriptorIds;
+import org.graylog2.contentpacks.model.entities.references.ValueReference;
+import org.graylog2.plugin.rest.ValidationResult;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotBlank;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@AutoValue
+@JsonTypeName(SlackEventNotificationConfig.TYPE_NAME)
+@JsonDeserialize(builder = SlackEventNotificationConfig.Builder.class)
+public abstract class SlackEventNotificationConfig implements EventNotificationConfig {
+	public static final String TYPE_NAME = "slack-notification-v1";
+
+	static final String FIELD_COLOR = "color";
+	static final String FIELD_WEBHOOK_URL = "webhook_url";
+	static final String FIELD_CHANNEL = "channel";
+	static final String FIELD_CUSTOM_MESSAGE = "custom_message";
+	static final String FIELD_USER_NAME = "user_name";
+	static final String FIELD_NOTIFY_CHANNEL = "notify_channel";
+	static final String FIELD_LINK_NAMES = "link_names";
+	static final String FIELD_ICON_URL = "icon_url";
+	static final String FIELD_ICON_EMOJI = "icon_emoji";
+
+
+	@JsonProperty(FIELD_COLOR)
+	@NotBlank
+	public abstract String color();
+
+	@JsonProperty(FIELD_WEBHOOK_URL)
+	@NotBlank
+	public abstract String webhookUrl();
+
+	@JsonProperty(FIELD_CHANNEL)
+	@NotBlank
+	public abstract String channel();
+
+	@JsonProperty(FIELD_CUSTOM_MESSAGE)
+	public abstract String customMessage();
+
+	@JsonProperty(FIELD_USER_NAME)
+	@Nullable
+	public abstract String userName();
+
+	@JsonProperty(FIELD_NOTIFY_CHANNEL)
+	public abstract boolean notifyChannel();
+
+	@JsonProperty(FIELD_LINK_NAMES)
+	public abstract boolean linkNames();
+
+	@JsonProperty(FIELD_ICON_URL)
+	@Nullable
+	public abstract String iconUrl();
+
+	@JsonProperty(FIELD_ICON_EMOJI)
+	@Nullable
+	public abstract String iconEmoji();
+
+
+	@Override
+	@JsonIgnore
+	public JobTriggerData toJobTriggerData(EventDto dto) {
+		return EventNotificationExecutionJob.Data.builder().eventDto(dto).build();
+	}
+
+	public static SlackEventNotificationConfig.Builder builder() {
+		return SlackEventNotificationConfig.Builder.create();
+	}
+
+	@Override
+	@JsonIgnore
+	public ValidationResult validate() {
+		ValidationResult validation =  new ValidationResult();
+
+		final String regex = "https:\\/\\/hooks.slack.com\\/services\\/";
+		final Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+		final Matcher matcher = pattern.matcher(webhookUrl());
```

Comment:
Having the regex `Pattern` be compiled every time `validate()` is called eliminates the efficiency of compiling the pattern.  The `regex` and `pattern` are not going to change from one call of `validate()` to another and thus you should consider promoting them up to class-level `static final` attributes